zip-0226.rst

```diff
@@ -9,7 +9,480 @@
+   T.4d: flagsOrchard                        (1 byte)
+   T.4e: valueBalanceOrchard                 (64-bit signed little-endian)  
+   T.4f: anchorOrchard                       (32 bytes)
+
+T.4a: orchard_actions_compact_digest
+''''''''''''''''''''''''''''''''''''
+
+A BLAKE2b-256 hash of the subset of Orchard Action information intended to be included in
+an updated version of the ZIP-307 [#zip-0307]_ ``CompactBlock`` format for all Orchard
+Actions belonging to the transaction. For each Action, the following elements are included
+in the hash::
+
+   T.4a.i  : nullifier            (field encoding bytes)
+   T.4a.ii : cmx                  (field encoding bytes)
+   T.4a.iii: ephemeralKey         (field encoding bytes)
+   T.4a.iv : encCiphertext[..84]  (First 84 bytes of field encoding)  [UPDATED FOR ZSA]
+
+The personalization field of this hash is the same as in ZIP 244::
+
+  "ZTxIdOrcActCHash"
+
+
+T.4b: orchard_actions_memos_digest
+''''''''''''''''''''''''''''''''''
+
+A BLAKE2b-256 hash of the subset of Orchard shielded memo field data for all Orchard
+Actions belonging to the transaction. For each Action, the following elements are included
+in the hash::
+
+    T.4b.i: encCiphertext[84..596] (contents of the encrypted memo field)  [UPDATED FOR ZSA]
+
+The personalization field of this hash remains identical to ZIP 244::
+
+  "ZTxIdOrcActMHash"
+
+
+T.4c: orchard_actions_noncompact_digest
+'''''''''''''''''''''''''''''''''''''''
+
+A BLAKE2b-256 hash of the remaining subset of Orchard Action information **not** intended
+for inclusion in an updated version of the the ZIP 307 [#zip-0307]_ ``CompactBlock``
+format, for all Orchard Actions belonging to the transaction. For each Action,
+the following elements are included in the hash::
+
+   T.4d.i  : cv                    (field encoding bytes)
+   T.4d.ii : rk                    (field encoding bytes)
+   T.4d.iii: encCiphertext[596..]  (post-memo suffix of field encoding)  [UPDATED FOR ZSA]
+   T.4d.iv : outCiphertext         (field encoding bytes)
+
+The personalization field of this hash is defined identically to ZIP 244::
+
+    "ZTxIdOrcActNHash"
+
+T.5: issuance_digest
+````````````````````
+The details of the computation of this value are in ZIP 227 [#zip-0227-txiddigest]_.
+
+Signature Digest and Authorizing Data Commitment
+================================================
+
+The details of the changes to these algorithms are in ZIP 227 [#zip-0227-sigdigest]_ [#zip-0227-authcommitment]_.
+
+Security and Privacy Considerations
+===================================
+
+- The notes generated via the ZSA protocol are different from those generated via the Orchard protocol. As with any protocol upgrade, it will be possible to distinguish between notes generated by each protocol. However, all ZEC notes will be fully spendable with the ZSA protocol transaction structure due to the built-in backward compatibility.
```

Comment:
This could be slightly misleading. The transaction version is distinguishable, but the notes are not (absent the required viewing keys).

Reply:
Edited this portion.